File: Backend/src/Application/Manutencoes/GetStatus/ManutencaoPendenciaResponse.cs
using Domain.Common;
using Domain.Manutencoes;

namespace Application.Manutencoes.GetStatus;

public class ManutencaoPendenciaResponse
{
    public Guid Id { get; set; }
    public string Nome { get; set; }
    public NivelAlertaEnum Status { get; set; }
    public DateOnly? DataVencimento { get; set; }
    public int? DiasRestantes { get; set; }
    public int? OdometroVencimento { get; set; }
    public int? KmRestantes { get; set; }

    public ManutencaoPendenciaResponse(Manutencao manutencao, DateOnly hoje, int? odometroAtual)
    {
        Id = manutencao.Id;
        Nome = manutencao.Nome;
        Status = manutencao.CalcularStatus(hoje, odometroAtual);
        DataVencimento = manutencao.DataVencimento;
        DiasRestantes = manutencao.CalcularDiasRestantes(hoje);
        OdometroVencimento = manutencao.OdometroVencimento;
        KmRestantes = manutencao.CalcularKmRestantes(odometroAtual);
    }
}

File: Frontend/src/screens/manutencao/api/types/ManutencaoPendenciaResponse.ts
import type { NivelAlertaEnum } from "@/screens/common/api/types/NivelAlertaEnum";

export interface ManutencaoPendenciaResponse {
  id: string;
  nome: string;
  status: NivelAlertaEnum;
  dataVencimento: string | null;
  diasRestantes: number | null;
  odometroVencimento: number | null;
  kmRestantes: number | null;
}
